flowzone.yml

```diff
@@ -55,9 +55,9 @@
       decimal_sha=$(echo "ibase=16; $commit_sha" | bc)
       version_tag="${version}-dev${decimal_sha}"
 
-      echo "package=${package}" >> $GITHUB_OUTPUT
-      echo "version=${version}" >> $GITHUB_OUTPUT
-      echo "version_tag=${version_tag}" >> $GITHUB_OUTPUT
+      echo "package=${package}" >> "${GITHUB_OUTPUT}"
+      echo "version=${version}" >> "${GITHUB_OUTPUT}"
+      echo "version_tag=${version_tag}" >> "${GITHUB_OUTPUT}"
```

Comment:
Did you have to manually fix all of these linter errors in your editor? Did you use the VSCode plugin?
There [doesn't appear to be](https://github.com/rhysd/actionlint/issues/358) a `-fix` option for actionlint.

Reply:
I don't use VSCode period. I use basic (sometimes grep) search/replace in BBEdit.

Reply:
@klutchell I must be blind, but I can't see where we are now ignoring checks for unquoted variables (npm run lint)...

Reply:
As of my [last PR](https://github.com/product-os/flowzone/pull/977/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R7), we are not ignoring checks for unquoted variables in `npm run lint`.

But shellcheck is still disabled in the action until we get an idea of how many repos will break if enforced.
https://github.com/product-os/flowzone/blob/014534c836dee687edd773ed84baea940995fc32/flowzone.yml#L1210-L1211

Reply:
OK, something isn't right then:

```
$ git branch
* ab77/release-notes

$ git rebase master
Current branch ab77/release-notes is up to date.

$ actionlint .github/workflows/flowzone.yml; echo $?
0

$ cat foo.sh 
#!/usr/bin/env sh

FOO=bar

echo $foo

$ shellcheck foo.sh 

In foo.sh line 3:
FOO=bar
^-^ SC2034 (warning): FOO appears unused. Verify use (or export if used externally).


In foo.sh line 5:
echo $foo
     ^--^ SC2154 (warning): foo is referenced but not assigned.
     ^--^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
echo "$foo"
...
```

Reply:
Actionlint does not process all files in the repository, only workflow files in `.github/workflows/*.yml`.

Were you expecting `npm run lint` to process `foo.sh`?

Reply:
I am expecting it to pick up unquoted env vars in `.github/workflows/flowzone.yml` and from what I can see, it passed them currently.

Reply:
Can you link to these unquoted vars in your PR so I can see what was skipped? What if you introduce a different shellcheck error?

Reply:
Even invalid shell passes here, it might be a local workstation issue - can you confirm it flags an unquoted env var on your end?

Reply:
Can confirm it works on my laptop.

```yml
      - name: Toggle auto-merge
        if: steps.is_draft_pr.outputs.result == 'false'
        env:
          <<: *gitHubCliEnvironment
          # DO NOT use the automatic github token (GITHUB_TOKEN) as it will not trigger merge events!
          GH_TOKEN: "${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}"
        # As long as we avoid the --admin flag, we should never bypass branch protections
        # See: https://cli.github.com/manual/gh_pr_merge
        run: |
          gh pr merge ${{ github.event.pull_request.number }} --merge --auto || true
          # add a shellcheck lint error
          foo=$(uname -a)
          echo $foo
```
```text
> npm run build && npm run lint                                                   

> flowzone@11.1.9 build
> node build


> flowzone@11.1.9 lint
> docker run --rm -v $(pwd):/repo --workdir /repo rhysd/actionlint:1.6.27 -color

.github/workflows/flowzone.yml:4246:9: shellcheck reported issue in this script: SC2086:info:3:6: Double quote to prevent globbing and word splitting [shellcheck]
     |
4246 |         run: |
     |  
```

Make sure that the lint errors you are introducing are present in `.github/workflows/flowzone.yml` since that's what's being checked. That file is updated with `npm run build` or the pre-commit hooks.